force-app/main/default/classes/GooglePlacesUtility.cls

```diff
@@ -47,6 +69,31 @@ public with sharing class GooglePlacesUtility {
 
         return response;
     }
+    */
+
+    // Method for creating API call for both textsearch and details call
+    public static HttpResponse createApiCall(String callType, Map<String, String> callParameters) {
```

Comment:
@Julian88Tex Re-factored the createTextSearch and createDetailsCall methods into one method that can do both. Also validating and checking for errors from this method since previously it was checking for status codes and error messages all over the place